test_matching.py

```diff
@@ -10,16 +10,12 @@
 class KnockServerTestCase(unittest.TestCase):
 
     def setUp(self):
-        self.db_fd, app.config['DATABASE'] = tempfile.mkstemp()
-        app.config['TESTING'] = True
         self.app = app.test_client()
-        with app.app_context():
-            database.init_db()
+        database.init_db()
 
     def tearDown(self):
         db_session.remove()
-        os.close(self.db_fd)
-        os.unlink(app.config['DATABASE'])
+        database.silent_remove_test_db()
 
```

Comment:
Forcibly remove the DB at the end of each test.  I know that the database file does this, but we need it to happen once at the end of testing as well.